### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by berylcl**
In this PR, I included
  - Create class Person with the following:
       - Instance vars: @id, @name, and @age.
       - Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
      -  Getters for @id, @name, and @age.
      -  Setters for @name and @age.
       - Private method of_age? that returns true if @age is greater or equal to 18 and false otherwise.
       - Public method can_use_services? that returns true if person is of age or if they have permission from parents.
 - Create class Student with the following:
       - Inherits from Person.
       - Constructor extends parent's constructor by adding @classroom and a parameter for it.
       - Method play_hooky that returns "¯\(ツ)/¯".
-  Create class Teacher with the following:
       - Inherits from Person.
       - Constructor extends parent's constructor by adding @specialization and a parameter for it.
      -  Override can_use_services? so it always returns true.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# FrozenStringLiteral: true
+
+class Person
+  attr_reader :id, :name, :age
+
```
Comment (Owusu-Desmond) [inline person.rb]: - [ ] Please make sure that the `:name` and `:age` instance variables should be able to be modified so I suggest they are defined as attr_accessors (both Getters and Setters are defined) :pray:

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  private
+
+  def of_age?
+    @age >= 18
+  end
+
+  def generate_id
+    rand(1000..9999)
+  end
+end
```
Comment (Owusu-Desmond) [inline person.rb:26]: - [ ] From the project requirement, you should create a public method `can_use_services?` that returns true if the person is of age or if they have permission from parents :blush:

File: README.md [hunk 3]
```diff
@@ -1 +1,158 @@
...
+### Prerequisites
+
+In order to run this project you need:
+
+- An IDE e.g [Visual studio code](https://code.visualstudio.com/).
+- [A terminal](https://code.visualstudio.com/docs/terminal/basics).
+
+### Setup
+### Setup
+
```
Comment (Owusu-Desmond) [inline README.md]: - [ ] You have the`setup` section repeated in your readme file so make sure you remove one to make your readme file professional :blush:

File: README.md [hunk 4]
```diff
@@ -1 +1,158 @@
...
+```
+rubocop
+```
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+
+## 👤 Author <a name="author"></a>
+
+## 👤 Author <a name="author"></a>
```
Comment (Owusu-Desmond) [inline README.md]: - [ ] You have the`Author` section repeated in your readme file so make sure you remove one to make your readme file professional :blush:

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @berylcl , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tresorsawasawa — APPROVED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @berylcl 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is professional :heavy_check_mark: 

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**